File: content/examples/other/leetcode/problems/meeting-rooms/example-2/solution.js
function canAttendMeetingsTwoArrays(intervals) {
  const starts = intervals.map((i) => i[0]).sort((a, b) => a - b);
  const ends = intervals.map((i) => i[1]).sort((a, b) => a - b);
  for (let i = 1; i < starts.length; i += 1) {
    if (starts[i] < ends[i - 1]) return false;
  }
  return true;
}

if (require.main === module) {
  console.log(canAttendMeetingsTwoArrays([[0, 30], [5, 10], [15, 20]]));
  console.log(canAttendMeetingsTwoArrays([[7, 10], [2, 4]]));
}

module.exports = { canAttendMeetingsTwoArrays };

File: content/examples/other/leetcode/problems/meeting-rooms/example-2/Solution.java
import java.util.Arrays;

public class Solution {
  static boolean canAttendMeetingsTwoArrays(int[][] intervals) {
    int n = intervals.length;
    int[] starts = new int[n];
    int[] ends = new int[n];
    for (int i = 0; i < n; i += 1) {
      starts[i] = intervals[i][0];
      ends[i] = intervals[i][1];
    }
    Arrays.sort(starts);
    Arrays.sort(ends);
    for (int i = 1; i < n; i += 1) {
      if (starts[i] < ends[i - 1]) return false;
    }
    return true;
  }

  public static void main(String[] args) {
    System.out.println(canAttendMeetingsTwoArrays(new int[][] {{0, 30}, {5, 10}, {15, 20}}));
    System.out.println(canAttendMeetingsTwoArrays(new int[][] {{7, 10}, {2, 4}}));
  }
}
